cheroot/test/test_ssl.py

```diff
@@ -375,6 +409,108 @@ def test_tls_client_auth(
         assert any(e in err_text for e in expected_substrings)
 
 
+@pytest.mark.parametrize(
+    'adapter_type',
+    (
+        'builtin',
+        'pyopenssl',
+    ),
+)
+@pytest.mark.parametrize(
+    'tls_verify_mode',
+    (
+        ssl.CERT_NONE,  # server shouldn't validate client cert
+        ssl.CERT_OPTIONAL,  # same as CERT_REQUIRED in client mode, don't use
+        ssl.CERT_REQUIRED,  # server should validate if client cert CA is OK
+    ),
+)
+def test_ssl_env(
+    mocker,
+    tls_wsgi_server, adapter_type,
+    ca, tls_verify_mode, tls_certificate,
+    tls_certificate_chain_pem_path,
+    tls_certificate_private_key_pem_path,
+    tls_ca_certificate_pem_path,
+):
+    """Test the SSL environment generated by the SSL adapters."""
+    interface, _host, port = _get_conn_data(ANY_INTERFACE_IPV4)
+
+    with mocker.mock_module.patch(
+        'idna.core.ulabel',
+        return_value=ntob('127.0.0.1'),
+    ):
+        client_cert = ca.issue_cert(ntou('127.0.0.1'),)
+
+    with client_cert.private_key_and_cert_chain_pem.tempfile() as cl_pem:
+        tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+        tls_adapter = tls_adapter_cls(
+            tls_certificate_chain_pem_path,
+            tls_certificate_private_key_pem_path,
+        )
+        if adapter_type == 'pyopenssl':
+            tls_adapter.context = tls_adapter.get_context()
+            tls_adapter.context.set_verify(
+                _stdlib_to_openssl_verify[tls_verify_mode],
+                lambda conn, cert, errno, depth, preverify_ok: preverify_ok,
+            )
+        else:
+            tls_adapter.context.verify_mode = tls_verify_mode
+
+        ca.configure_trust(tls_adapter.context)
+        tls_certificate.configure_cert(tls_adapter.context)
+
+        tlswsgiserver = tls_wsgi_server((interface, port), tls_adapter)
+
+        interface, _host, port = _get_conn_data(tlswsgiserver.bind_addr)
+
+        resp = requests.get(
+            'https://' + interface + ':' + str(port) + '/',
+            verify=tls_ca_certificate_pem_path,
+            cert=cl_pem,
+        )
+        if PY34 and resp.status_code != 200:
+            pytest.xfail(
+                'Python 3.4 has problems with verifying client certs',
+            )
+
+        env = pickle.loads(resp.content)
+
+        # hard coded env
+        assert 'wsgi.url_scheme' in env
+        assert env['wsgi.url_scheme'] == 'https'
+        assert 'HTTPS' in env
+        assert env['HTTPS'] == 'on'
+
+        # ensure these are present
+        for key in {'SSL_VERSION_INTERFACE', 'SSL_VERSION_LIBRARY'}:
+            assert key in env
+
+        # pyOpenSSL generates the env before the handshake completes
+        if adapter_type == 'pyopenssl':
+            return
+
+        for key in {'SSL_PROTOCOL', 'SSL_CIPHER'}:
+            assert key in env
+
+        # client certificate env
+        if tls_verify_mode == ssl.CERT_NONE:
+            assert 'SSL_CLIENT_VERIFY' in env
```

Comment:
This is unnecessary. This check is already embedded into the next assertion line.

Reply:
removed